server/src/main/java/org/elasticsearch/index/mapper/IdFieldMapper.java

```diff
@@ -160,9 +169,6 @@ public Query termsQuery(List<?> values, QueryShardContext context) {
 
         @Override
         public IndexFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName) {
-            if (indexOptions() == IndexOptions.NONE) {
-                throw new IllegalArgumentException("Fielddata access on the _id field is disallowed");
```

Comment:
have we changed behaviour here?

Reply:
No, the ID field is always indexed (you can't configure metadata).